task_03/script.js

```diff
@@ -0,0 +1,227 @@
+const gameResult = document.querySelector('.game__result');
+const gameForm = document.querySelector('.game__form');
+const formField = document.querySelector('.form__field');
+const formLabel = document.querySelector('.form__label');
+const buttonSubmit = document.querySelector('.button--submit');
+const sidebarList = document.querySelector('.sidebar-list');
+const tableCells = document.querySelectorAll('.board__box');
+const userInput = document.getElementById('username');
+const formErrorMessage = document.getElementById('form-error-message');
+
+const COMBINATIONS = {
+	0: [[1, 2], [3, 6], [4, 8]],
+	1: [[0, 2], [4, 7]],
+	2: [[0, 1], [5, 8], [4, 6]],
+	3: [[0, 6], [4, 5]],
+	4: [[3, 5], [1, 7], [0, 8], [2, 6]],
+	5: [[3, 4], [2, 8]],
+	6: [[0, 3], [7, 8], [2, 4]],
+	7: [[1, 4], [6, 8]],
+	8: [[6, 7], [2, 5], [0, 4]],
+};
+
+let numberOfMoves = 0;
+let gameActive = true;
+let currentPlayer = 'X';
+let gameState = ['', '', '', '', '', '', '', '', ''];
+
+const handlePlayerChange = () => currentPlayer = currentPlayer === 'X' ? 'O' : 'X';
+const clearUserInput = () => userInput.value = '';
+const winningMessage = () => `Player ${currentPlayer} has won in ${numberOfMoves} moves!`;
+const drawMessage = () => `Game ended in a draw!`;
+
+const handleCellPlayed = (clickedCell, clickedCellIndex) => {
+	gameState[clickedCellIndex] = currentPlayer;
+	clickedCell.innerHTML = currentPlayer;
+	numberOfMoves++;
+};
+
+const handleResultValidation = (clickedCellIndex) => {
+	let roundWon = false;
+	const combinations = COMBINATIONS[clickedCellIndex];
+
+	combinations.forEach(([firstIndex, secondIndex]) => {
+		if (gameState[firstIndex] === currentPlayer && gameState[secondIndex] === currentPlayer) {
+			roundWon = true;
+		}
+	});
+
+	if (roundWon) {
```

Comment:
Extract into `handleWinning(player)` function and call it on line 46. That also removes the need for roundWon and roundDraw variables.